### ISSUE TITLE
OR Modifier Module

### ISSUE THREAD

**Issue by jonfroehlich** [myjeeed/MakerWear#59]
The original Google Doc for module ideas had some boolean logic like ORs. Did you guys ever revisit this? If not, let's start the discussion here. If so, let's summarize the discussion here.

In our current architecture, you get ANDs implicitly by cascading modules (though obviously we don't ever use or surface that boolean language); however, there is really no current way to get OR behavior, right?

In thinking about how to do this, it seems like the simplest solution would be a module that has two inputs and one output where Output = Input1 | Input2 (could do three outputs to be consistent with other modules but this could also be confusing and then we would have five sides taken up rather than the standard four). 

There are two major complications with this: (i) the socket mesh assumes only one input, so I believe we would have to update this (this seems like a huge change and not worth addressing until we switch over to next-generation digital architecture); (ii) our architecture is analog--so how do you OR analog values. For (i), I need feedback on whether our current architecture supports multiple inputs into a module (it would work in ReWear, for example, but maybe not in current MakerWear architecture).

For (ii), we could do a few thing:

Option 1 (Binary): we discretize the inputs into binary so that the output is also binary. This is the easiest solution. We basically decide a cutoff (e.g., midway voltage point) and treat all values below that as 0 and all above as 1.

Option 2 (Analog): we compute some function on the inputs that could be reasonably considered an OR (like maybe averaging the signals? Of course, if we do that, maybe this modifier would need to be called an Average).

Thoughts?

**Comment by alexjiao125** [comment]
Right now we do have a way of determining whether analog values are LOW or HIGH by checking if they are under 300 (low) or above 700 (high); this is implemented in our fade and counter modules.  There are enough pins on an ATtiny in order to read of the two input pins, determine if at least one input signal is HIGH, and output accordingly.  

One thing that would need changing is the bottom-layer PCB on the modules themselves to support 2 inputs and one output. You could also possibly use the same bottom-layer PCBs but rotate them around and use 2 of the 3 outputs as inputs and the lone input as your new output (Just a thought, probably has it's downsides).

**Comment by jonfroehlich** [comment]
Right, this was my first point (i)--and seems like a major impediment to implementing any multi-input module at moment. So, I think best to have this as FW for CHI paper then. :) We can come back to this for next generation architecture.
